### DISCUSSION TITLE
Scope store and work with SwiftUIX got wrong UI size

### DISCUSSION THREAD

**Discussion by marcuswu0814** [pointfreeco/swift-composable-architecture#2762] [Q&A] [answered]
![demo](https://github.com/pointfreeco/swift-composable-architecture/assets/6396757/41364306-68bc-4a44-8649-f355e210d33b)

I'm trying to use the `PaginationView` provide by [SwiftUIX](https://github.com/SwiftUIX/SwiftUIX?tab=readme-ov-file#pagination) library. But when I using the `ForEach(store.scope) ...` in the content builder, I got wrong size. It seems used a different initializer overload, and the page be considered a single view:

![CleanShot 2024-02-02 at 18 30 30@2x](https://github.com/pointfreeco/swift-composable-architecture/assets/6396757/44d631f3-5166-482f-828c-fa2a1b2b73c0)

If I just using normal `ForEach(store.pages)...`, it work but it's not what I want.

The other initializer:

![CleanShot 2024-02-02 at 18 35 01@2x](https://github.com/pointfreeco/swift-composable-architecture/assets/6396757/ea6eb6f1-eff9-4080-aa02-5b0de8d68667)

---

The reproduce project and detail description in [here](https://github.com/marcuswu0814/TCASwiftUIX). If anyone has encountered a similar problem, please give me some advice. 🙏

---

I also create a [issue](https://github.com/SwiftUIX/SwiftUIX/issues/467) at SwiftUIX.

**Comment by mbrandonw** [comment, marked as answer]
The issue seems to be in SwiftUIX. If you add `@ViewBuilder` to the `content` closure in the initializer you want, then it will work. You can either PR that to SwiftUIX, or you can drop this in your project:

```swift
extension PaginationView {
  public init<Data, ID>(
    axis: Axis = .horizontal,
    transitionStyle: UIPageViewController.TransitionStyle = .scroll,
    showsIndicators: Bool = true,
    @ViewBuilder // 👈 This is important
    content: () -> ForEach<Data, ID, Page>
  ) where Data.Element: Identifiable {
    self.init(
      content: .init(content()),
      axis: axis,
      transitionStyle: transitionStyle,
      showsIndicators: showsIndicators
    )
  }
}
```

Or if you don't want to use that extension you can be very explicit with your types:

```swift
  var body: some View {
    PaginationView(
      axis: .vertical,
      showsIndicators: false
    ) { () -> ForEach<Array<StoreOf<Page>>, ObjectIdentifier, PageView> in
      ForEach(Array(store.scope(state: \.pages, action: \.pages))) { store in
        PageView(store: store)
      }
    }
  }
```

**Reply by marcuswu0814** [reply]
Hi @mbrandonw, it work like a charm, thanks for your quick help, it save me lots of time. 😊

I've also created a [PR](https://github.com/SwiftUIX/SwiftUIX/pull/468) at SwiftUIX, hoping it can assist anyone in the future.

**Reply by vmanot** [reply]
@mbrandonw @marcuswu0814 thank you so much for the PR! Just merged it in. I'll be cutting a new release later this week, feel free to use the tip of `master` for now.
